nmostesting/suites/IS0404Test.py

```diff
@@ -0,0 +1,288 @@
+        # * The Sender which was put 'offline' will then come back online at a random moment within
+        #   the following 60 seconds.
+        # * As soon as the NCuT detects the Sender has come back online the user must press the
+        #   'Next' button on the Testing Façade.
+        # * The button must be pressed within 30 seconds of the Sender being put back 'online'.
+        #   This includes any latency between the Sender being put 'online' and the NCuT updating.
+
+        try:
+            # Check senders
+
+            question = 'The NCuT should be able to discover and dynamically update all the Senders ' \
+                'that are registered in the Registry.\n\n' \
+                'Use the NCuT to browse and take note of the Senders that are available.\n\n' \
+                'After the \'Next\' button has been clicked one of those senders will be put \'offline\'.'
+            possible_answers = []
+
+            self._invoke_testing_facade(question, possible_answers, test_type="action")
+
+            # Take one of the senders offline
+            possible_answers = [{'answer_id': 'answer_'+str(i), 'label': s['label'],
+                                 'description': s['description'], 'id': s['id'], 'answer_str': s['answer_str']}
+                                for i, s in enumerate(self.senders) if s['registered']]
+            answer_indices = [index for index, s in enumerate(self.senders) if s['registered']]
+            offline_sender_index = random.choice(answer_indices)
+            expected_answer = 'answer_' + str(offline_sender_index)
+
+            self._delete_sender(test, self.senders[offline_sender_index])
+
+            # Set the offline sender to registered false for future tests
+            self.senders[offline_sender_index]['registered'] = False
+
+            # Recheck senders
+            question = 'Please refresh your NCuT and select the sender which has been put \'offline\''
+
+            actual_answer = self._invoke_testing_facade(
+                question, possible_answers, test_type="single_choice", multipart_test=1)['answer_response']
+
+            if actual_answer != expected_answer:
+                return test.FAIL('Offline/online sender not handled: Incorrect sender identified')
+
+            max_time_until_online = 60
+            max_time_to_answer = 30
+
+            question = 'The sender which was put \'offline\' will come back online at a random moment ' \
+                'within the next ' + str(max_time_until_online) + ' seconds. ' \
+                'As soon as the NCuT detects the sender has come back online please press the \'Next\' button.\n\n' \
+                'The button must be pressed within ' + str(max_time_to_answer) + \
+                ' seconds of the Sender being put back \'online\'. \n\n' \
+                'This includes any latency between the Sender being put \'online\' and the NCuT updating.'
```

Comment:
Quick play with multi-line strings and using either `"... {} ... {} ...".format(a, b, ...)` or, now that we require Python 3.6, even f-strings (`"... {a} ... {b} ..."`):

```suggestion
            question = textwrap.dedent(f"""\
                       The sender which was put 'offline' will come back online at a random moment
                       within the next {max_time_until_online} seconds.
                       As soon as the NCuT detects the sender has come back online please press the 'Next' button.

                       The button must be pressed within {max_time_to_answer} seconds of the Sender being put back 'online'.

                       This includes any latency between the Sender being put 'online' and the NCuT updating.
                       """)
```

Reply:
`dedent` now generally used when embedding parameters in multi-line strings